packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -800,6 +801,21 @@ function prepareFreshStack(root, expirationTime) {
   }
 }
 
+// a wrapper around scheduleTimeout, that immediately
+// runs the callback when inside an act() scope
+function scheduleWorkTimeout(callback, period) {
+  if (flushSuspenseFallbacksInTests === true) {
```

Comment:
Can we go back to the previous version where you wrap each case of the switch statement with this check? Semantically it's *almost* the same, but not exactly. There are lots of subtle edge cases that happen in between. E.g. in the `RootSuspended` case, we sometimes restart instead of Suspended:

https://github.com/facebook/react/blob/55bc393f726ddf3def83ca5a43cfc38a45d2e421/packages/react-reconciler/src/ReactFiberWorkLoop.js#L1037-L1048

It's also less of a refactor hazard the next time someone edits this code.

Reply:
agreed, will do